test/randomized_benchmarking/rb_generate_data.py

```diff
@@ -107,36 +108,51 @@ def _generate_rb_fitter_data(dir_name: str, rb_exp_name: str, exp_attributes: di
         backend, noise_model=noise_model, basis_gates=transpiled_base_gate
     ).block_for_results()
     print("Done running experiment")
+    experiment_obj.block_for_results()
     exp_results = experiment_obj.data()
     with open(results_file_path, "w") as json_file:
         joined_list_data = [exp_attributes, exp_results]
         json_file.write(json.dumps(joined_list_data))
-    _analysis_save(
-        [res.data() for res in experiment_obj.analysis_results(None)], analysis_file_path
-    )
+    _analysis_save(experiment_obj, analysis_file_path, "RBAnalysis")
 
 
-def _analysis_save(analysis_data: list, analysis_file_path: str):
+def _analysis_save(experiment_obj, analysis_file_path: str, analysis_type: str):
     """
     The function is creating a json file from the data of the RB experiment analysis.
     Args:
-        analysis_data (list): The data from the analysis of the experiment.
+        analysis_data: The data from the analysis of the experiment.
         analysis_file_path (str): The path to save the json file.
+        analysis_type (str): The type of analysis class used
     """
-    samples_analysis_list = []
-    for sample_analysis in analysis_data:
-        sample_analysis["popt"] = list(sample_analysis["popt"])
-        sample_analysis["popt_err"] = list(sample_analysis["popt_err"])
-        sample_analysis["pcov"] = list(sample_analysis["pcov"])
-        for idx, item in enumerate(sample_analysis["pcov"]):
-            sample_analysis["pcov"][idx] = list(item)
-        if "EPG" in sample_analysis:
-            epg_keys = list(sample_analysis["EPG"].keys())
-            for qubits in epg_keys:
-                sample_analysis["EPG"][str(qubits)] = sample_analysis["EPG"].pop(qubits)
-        samples_analysis_list.append(sample_analysis)
+    analysis_json_dict = {}
+    for result in experiment_obj.analysis_results():
+        if result.name == analysis_type:
```

Comment:
Now this has prefix since data points are also saved under the same name. Also some keys are changed, i.e. "pcov" in extra field is quite understandable. "popt_err" is not in here since the same information is stored in the value.
https://github.com/Qiskit/qiskit-experiments/blob/13473af89c5771527e76e1447e22e65b78e8848b/qiskit_experiments/curve_analysis/curve_analysis.py#L1041-L1053